### ISSUE TITLE
parameter-dependent problems where parameter appears in a boundary condition

### ISSUE THREAD

**Issue by hrothgar** [chebfun/chebfun#1209]
I am having no luck trying to solve a parameter-dependent problem in which the parameter appears in the boundary condition. Is there a way to get it to work, or is this feature not supported? I see no reason why it shouldn’t work.
```
op = @(x,u,p) diff(u,2) + u;
bc = @(x,u,p) [ u(0)
                u(1) - 2
                feval(diff(u),0) - p ];
A = chebop(op, [0 1], bc);
sol = A\0
```
I get the following error:
```
Error using cat
Dimensions of matrices being concatenated are not consistent.
Error in cell2mat (line 87)
        m = cat(2,m{:});
Error in chebDiscretization/matrix (line 63)
        M = cell2mat(A);
...
```
Passing an initial guess with correct dimensions does not help:
```
op = @(x,u,p) diff(u,2) + u;
bc = @(x,u,p) [ u(0)
                u(1) - 2
                feval(diff(u),0) - p ];

x = chebfun('x', [0 1])
init = [0*x; 0];

A = chebop(op, [0 1], bc, init);
sol = A\0
```

**Comment by asgeirbirkis** [comment]
Here are some more lines of code related to this issue that came from the help mailing list:
```
%% Example 1 (fine)
L = chebop(@(t,x,p) diff(x)-1,[-1 1]);
L.lbc = @(x,p) x;
L.rbc = @(x,p) x-p;
X = L\0;
[x, p] = deal(X);
plot(x)
p
%Clearly, the solution is p=2 and x(t) is a straight line from (-1,0) to (1,2). 
% So far, Chebfun works fine. But now we try the same problem but using .bc syntax:
%% Example 2 (not working)
L = chebop(@(t,x,p) diff(x)-1,[-1 1]);
L.bc = @(t,x,p) [x(-1);x(1)-p];
X = L\0;
[x, p] = deal(X);
plot(x)
p
%Chebfun fails with a complicated error message, which I do not understand. However, if I slightly %change the problem so that the parameter p appears in the operator (replace -1 by -p in the first 
%line) and in turn removing the p from the bc (replace -p by -2 in the second line), then Chebfun 
%again works fine and gives the same x(t) and p=1 as expected:
%% Example 3 (fine)
L = chebop(@(t,x,p) diff(x)-p,[-1 1]);
L.bc = @(t,x,p) [x(-1);x(1)-2];
X = L\0;
[x, p] = deal(X);
plot(x)
p
```

**Comment by asgeirbirkis** [comment]
Closed via #1418.
